docs/components.rst

```diff
@@ -45,28 +45,41 @@ Well, in Discord, clicking buttons and using slash commands are called ``interac
 Responding to interactions
 __________________________
 
-When responding, you have 2 choices in how you handle interactions. You can either wait for them in the command itself, or listen for them in a global event handler (similar to :meth:`on_slash_command_error`)
+When responding, you have 3 choices in how you handle interactions. You can either wait for them in the command itself, or listen for them in a global event handler (similar to :func:`on_slash_command_error`), or register async function as callback callback.
 
-Lets go through the most common method first, responding in the event itself. We simply need to :meth:`wait_for` the event, just like you do for reactions. For this we're going to use :func:`wait_for_component() <discord_slash.utils.manage_components>`, and we're going to only wait for events from the action row we just .
+Lets go through the most common method first, responding in the event itself. We simply need to :func:`wait_for` the event, just like you do for reactions. For this we're going to use :func:`wait_for_component() <discord_slash.utils.manage_components>`, and we're going to only wait for events from the action row we just sent.
 This method will return a :class:`ComponentContext <discord_slash.context.ComponentContext>` object that we can use to respond. For this example, we'll just edit the original message (:meth:`edit_origin() <discord_slash.context.ComponentContext.edit_origin>`)
 
 .. code-block:: python
 
     await ctx.send("My Message", components=[action_row])
     # note: this will only catch one button press, if you want more, put this in a loop
-    button_ctx: ComponentContext = await manage_components.wait_for_component(bot, action_row)
+    button_ctx: ComponentContext = await manage_components.wait_for_component(bot, components=action_row)
     await button_ctx.edit_origin(content="You pressed a button!")
 
 .. note:: It's worth being aware that if you handle the event in the command itself, it will not persist reboots. As such when you restart the bot, the interaction will fail
 
-Next we'll go over the alternative, a global event handler. This works just the same as :meth:`on_slash_command_error` or `on_ready`.
+Next we'll go over the alternative, a global event handler. This works just the same as :func:`on_slash_command_error` or `on_ready`. But note that this code will be triggered on any components interaction.
```

Comment:
No, neither `on_ready` nor `on_slash_command_error` is a class. I made it a function because it's not actually attached to class, it can be just any function registered as callback for this event